sourcemash/frontend/static/js/views/register_modal_view.js

```diff
@@ -0,0 +1,41 @@
+Sourcemash.Views.RegisterModalView = Backbone.View.extend({
+    template: JST['register-modal'],
+
+    events: {
+        'submit #register': 'registerFromModal',
+    },
+
+    registerFromModal: function(e) {
+        e.preventDefault();
+        var formData = JSON.stringify($("#register").serializeObject());
+        $('#register-password').val('');
+        $('#register-password-confirm').val('');
+        var posting = $.ajax({
+                          type: "POST",
+                          url: "/register",
+                          data: formData,
+                          success: this.showErrors,
```

Comment:
And no, the errors need to be extracted from the JSON response and directly rendered to the modal if we want to see them. 

The stuff we added last time allows the JSON message to be constructed before being sent, not parsed when it's received.